src/macros.jl

```diff
@@ -55,6 +55,231 @@ macro col(kw)
     esc(fun_to_vec(kw))
 end
 
+##############################################################################
+##
+## @byrow
+##
+##############################################################################
+"""
+Broadcast operations within DataFramesMeta.jl macros.
+
+`@byrow` is not a "real" Julia macro but rather serves as a "flag"
+to indicate that the anonymous function created by DataFramesMeta
+to represent an operation should be applied "by-row".
+
+If an expression starts with `@byrow`, either of the form `@byrow = f(:x)`
+in transformations or `@byrow f(:x)` in `@orderby`, `@where`, and `@with`,
+then the anonymous function created by DataFramesMeta is wrapped in the
+function-wrapped `DataFrames.ByRow`.
+
+### Examples
+
+```julia
+julia> df = DataFrame(a = [1, 2, 3, 4], b = [5, 6, 7, 8]);
+
+julia> @transform(df, @byrow c = :a * :b)
+4×3 DataFrame
+ Row │ a      b      c
+     │ Int64  Int64  Int64
+─────┼─────────────────────
+   1 │     1      5      5
+   2 │     2      6     12
+   3 │     3      7     21
+   4 │     4      8     32
+
+julia> @where(df, @byrow :a == 1 ? true : false)
+1×2 DataFrame
+ Row │ a      b
+     │ Int64  Int64
+─────┼──────────────
+   1 │     1      5
+```
+
+To avoid writing `@byrow` multiple times when performing multiple
+operations, it is allowed to use`@byrow` at the beginning of a block of
+operations. All transformations in the block will operate by row.
+
+```julia
+julia> @where df @byrow begin
+           :a > 1
+           :b < 5
+       end
+1×2 DataFrame
+ Row │ a      b
+     │ Int64  Int64
+─────┼──────────────
+   1 │     2      4
+```
+
+### Comparison with `@eachrow`
+
+To re-cap, the `@eachrow` rougly transforms
+
+```julia
+@eachrow df begin
+    :a * :b
+end
+```
+
+to
+
+```julia
+begin
+    function tempfun(a, b)
+        for i in eachindex(a)
+            a[i] * b[i]
+        end
+    end
+    tempfun(df.a, df.b)
+    df
+end
+```
+
+The function `*` is applied by-row. But the result of those operations
+is not stored anywhere, as with `for`-loops in Base Julia.
+Rather, `@eachrow` and `@eachrow!` return data frames.
+
+Now consider `@byrow`. `@byrow` transforms
+
+```julia
+@with df @byrow begin
+    :a * :b
+end
+```
+
+to
+
+```julia
+tempfun(a, b) = a * b
+tempfun.(df.a, df.b)
+```
+
+In contrast to `@eachrow`, `@with` combined with `@byrow` returns a vector of the
+broadcasted multiplication and not a data frame.
+
+Additionally, `@eachrow` and `@eachrow!` allow modifying a data
+data frame. Just as with Base Julia broadcasting, `@byrow` will
+not update columns.
```

Comment:
```suggestion
```suggestion 
Additionally, transformations applied using `@eachrow!` modify the input
data frame. On the contrary, `@byrow` does not update columns.
```
```